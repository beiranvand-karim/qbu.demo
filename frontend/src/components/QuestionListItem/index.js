import React from 'react'
import FlexWrapper from "../../StyledComponents/FlexWrapper"
import Question from "../../StyledComponents/Question"
import {Button, Jumbotron} from "reactstrap"
import {withRouter} from 'react-router-dom'

const navigateQuestionDetail = (props) => {
   props.history.push(`/questionDetail/${props.question.id}`)
};

export const QuestionListItem = (props) => {
   return <FlexWrapper>
      <Question>
         <Jumbotron>
            <h1 className="display-6">{props.question.title}</h1>
            <p className="lead">{props.question.username}</p>
            <p className="lead">{props.question.prize}</p>
            <hr className="my-2" />
            <p>{props.question.text}</p>
            <p className="lead">
               <Button color="primary" onClick={() => navigateQuestionDetail(props)}>visit question</Button>
            </p>
         </Jumbotron>
      </Question>
   </FlexWrapper>
};

export default withRouter(QuestionListItem)