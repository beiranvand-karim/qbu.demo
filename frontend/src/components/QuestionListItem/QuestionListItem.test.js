/* global describe, expect, it */
import React from 'react'
import {shallow} from 'enzyme'
import QuestionListItem from './'
import renderer from 'react-test-renderer'

describe('<QuestionListItem />', () => {
   const question = {
      "id": 8,
      "title": "who is the best soccer player?",
      "text": "we want to know who is curently the best soccer player ther is?",
      "prize": 2000,
      "userId": 7,
      "username": "admin"
   };
   it('should render without crashing', () => {
      shallow(<QuestionListItem question={question}/>)
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<QuestionListItem {...question}/>).toJSON();
      expect(tree).toMatchSnapshot()
   });
});