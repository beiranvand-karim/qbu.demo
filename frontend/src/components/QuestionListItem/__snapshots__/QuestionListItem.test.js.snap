// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionListItem /> should matches the snapshot 1`] = `
<div
  className="sc-bdVaJa gDhQkR"
>
  <div
    className="sc-bwzfXH eiaxpn"
  >
    <div
      className="jumbotron"
    >
      <h1
        className="display-6"
      >
        who is the best soccer player?
      </h1>
      <p
        className="lead"
      >
        admin
      </p>
      <p
        className="lead"
      >
        2000
      </p>
      <hr
        className="my-2"
      />
      <p>
        we want to know who is curently the best soccer player ther is?
      </p>
      <p
        className="lead"
      >
        <button
          aria-label={null}
          className="btn btn-primary"
          onClick={[Function]}
        >
          see all answers
        </button>
      </p>
    </div>
  </div>
</div>
`;
